third_party/libtock-drivers/src/nfc.rs

```diff
@@ -76,61 +76,53 @@ impl NfcTag {
     /// 1. Share with the driver a buffer.
     /// 2. Subscribe to having a successful receive callback.
     /// 3. Issue the request for reception.
-    pub fn receive(buf: &mut [u8; 256]) -> TransOrRecvStatus {
-        let result = syscalls::allow(DRIVER_NUMBER, allow_nr::RECEIVE, buf);
-        if result.is_err() {
-            return TransOrRecvStatus::InvalidBuffer;
-        }
-
+    pub fn receive(buf: &mut [u8; 256]) -> Result<RecvOp, TockError> {
+        let result = syscalls::allow(DRIVER_NUMBER, allow_nr::RECEIVE, buf)?;
+        // set callback with 2 arguments, to receive ReturnCode and RX Amount
         let done = Cell::new(false);
-        let mut alarm = || done.set(true);
-        let subscription = syscalls::subscribe::<callback::Identity0Consumer, _>(
+        let result_code = Cell::new(None);
+        let recv_amount = Cell::new(None);
+        let mut callback = |result, amount| {
+            result_code.set(Some(result));
+            recv_amount.set(Some(amount));
+            done.set(true)
```

Comment:
Did the underlying syscall interface change? Because the `callback` behaves differently now?

Reply:
No, just discovered how to send a callback with arguments 😅